File: backend/app/schemas/recommendation.py
from typing import List, Optional

from pydantic import BaseModel, Field

from app.schemas.movie import MovieOut


class RecommendationOut(BaseModel):
    movie: MovieOut
    score: float
    reason: str
    movie_genres: List[str] = Field(default_factory=list)
    matched_genres: List[str] = Field(default_factory=list)

    model_config = {"from_attributes": True}


class HomeRecommendationsOut(BaseModel):
    hero_message: str
    dominant_genre: Optional[str] = None
    mood: str
    recommendations: List[RecommendationOut]

File: android/app/src/main/java/com/app/movieit/data/model/Recommendation.kt
package com.app.movieit.data.model

import com.google.gson.annotations.SerializedName

data class RecommendationOut(
    val movie: Movie,
    val score: Float,
    val reason: String,
    @SerializedName("movie_genres") val movieGenres: List<String> = emptyList(),
    @SerializedName("matched_genres") val matchedGenres: List<String> = emptyList()
)

data class HomeRecommendationsOut(
    @SerializedName("hero_message") val heroMessage: String,
    @SerializedName("dominant_genre") val dominantGenre: String?,
    val mood: String,
    val recommendations: List<RecommendationOut>
)
